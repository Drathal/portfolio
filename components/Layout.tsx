import { FC } from 'react'
import Head from 'next/head'

import Header from './Header'

interface IProps {
  pageTitle: string
}

const Layout: FC<IProps> = ({ children, pageTitle }) => {
  return (
    <>
      <Head>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <title>{pageTitle}</title>
      </Head>
      <section className="layout">
        <Header />
        <div className="content">{children}</div>
      </section>
      <footer>Built by me!</footer>
    </>
  )
}

export default Layout
