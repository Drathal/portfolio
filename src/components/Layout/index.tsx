import { FC } from 'react'
import Head from 'next/head'
import Container from '@material-ui/core/Container'
import CssBaseline from '@material-ui/core/CssBaseline'
import { makeStyles } from '@material-ui/core/styles'

import Header from '../Header'
import Starfield from '../Starfield'
interface IProps {
  pageTitle: string
}

const useStyles = makeStyles(() => ({
  container: {
    marginTop: '2rem',
    position: 'relative'
  }
}))

const Layout: FC<IProps> = ({ children, pageTitle }) => {
  const classes = useStyles()
  return (
    <>
      <CssBaseline />
      <Starfield />

      <Head>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <title>{pageTitle}</title>
      </Head>

      <Header />

      <Container fixed className={classes.container}>
        {children}
      </Container>
    </>
  )
}

export default Layout
