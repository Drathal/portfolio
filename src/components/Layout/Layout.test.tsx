import { render } from '@testing-library/react'
import Layout from './index'

// render head elements into body
jest.mock('next/head', () => ({
  __esModule: true,
  default: ({ children }: { children: Array<React.ReactElement> }) => {
    return children
  }
}))

describe('<About />', () => {
  test('it can render', () => {
    const testValues = { pageTitle: 'title', child: 'child' }
    const { container, getByText } = render(
      <Layout pageTitle={testValues.pageTitle}>{testValues.child}</Layout>
    )

    expect(container).toMatchSnapshot()
    expect(getByText(testValues.child)).toBeInTheDocument()
    expect(getByText(testValues.pageTitle)).toBeInTheDocument()
  })
})
