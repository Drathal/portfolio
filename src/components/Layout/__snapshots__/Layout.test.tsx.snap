// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> it can render 1`] = `
<div>
  <div
    class="makeStyles-bg-2"
  >
    <div
      class="makeStyles-star1-3"
    />
    <div
      class="makeStyles-star2-4"
    />
    <div
      class="makeStyles-star3-5"
    />
  </div>
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <title>
    title
  </title>
  <nav
    class="nav"
  >
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-appBar-8 MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-10 MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root makeStyles-title-7 MuiTypography-h6"
          >
            <a
              href="/"
            >
              Markus Dethlefsen
            </a>
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/"
              >
                Timeline
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/about"
              >
                About
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      id="back-to-top-anchor"
    />
  </nav>
  <div
    class="makeStyles-rootZoom-9"
    role="presentation"
    style="transform: scale(0); visibility: hidden;"
  >
    <button
      aria-label="scroll back to top"
      class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiContainer-root makeStyles-container-1 MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    child
  </div>
  <footer />
</div>
`;
